vpn/tunnel.go

```diff
@@ -7,24 +7,36 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"net/netip"
 	"net/url"
 	"reflect"
 	"strconv"
 	"sync"
+	"time"
 	"unicode"
 
+	"golang.org/x/exp/maps"
 	"golang.org/x/xerrors"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
 	"tailscale.com/net/dns"
+	"tailscale.com/util/dnsname"
 	"tailscale.com/wgengine/router"
 
+	"github.com/google/uuid"
+
 	"cdr.dev/slog"
 	"github.com/coder/coder/v2/coderd/util/ptr"
 	"github.com/coder/coder/v2/tailnet"
 )
 
+// The interval at which the tunnel sends network status updates to the manager.
+const netStatusInterval = 30 * time.Second
```

Comment:
It would be nice not to have to poll here.  You can imagine someone triggering a workspace build, and then waiting for CoderVPN to acknowledge that the connection to it is up. 30s is kind of a long time for humans to wait.

I think this is fine for now, to get us rolling, but maybe cut it to 10s and we'll revisit whether we can get tailscale to kick us whenever the status changes on a peer.

Reply:
Ah yep, I realised this but accidentally changed it in the stacked PR instead of this one, will fix.

Reply:
> whether we can get tailscale to kick us whenever the status changes on a peer.
Yeah, this is definitely worth revisiting. I spent a little time on it when working on telemetry, but didn't get anywhere.